utils/middleware.py

```diff
@@ -0,0 +1,27 @@
+import logging
+from urllib.parse import urlsplit
+from fastapi import HTTPException, status
+
+from models.comment import CommentPostBody
+
+
+def remove_search_hash_from_url(featureUrl: str):
+    """
+    Removes the search part (starting with ?)
+    and the hash part (starting with #) from a given URL
+    """
+    try:
+        split_url = urlsplit(featureUrl)
+    except Exception:
+        logging.error("Feature URL middleware::Invalid feature URL")
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Invalid feature URL",
+        )
+    split_url._replace(query="", fragment="")
```

Comment:
The method `_replace()` returns a new SplitResult object, but the `remove_search_hash_from_url()` function does not return it or assign it to the original `split_url` object. To fix this, you can replace the line `split_url._replace(query="", fragment="")` with `split_url = split_url._replace(query="", fragment="")` to assign the new SplitResult object to the `split_url` variable. Alternatively, you can remove this line if it is unnecessary.
Please see the [doc](https://docs.python.org/dev/library/collections.html#collections.somenamedtuple._replace).
```suggestion
    split_url = split_url._replace(query="", fragment="")
```

Reply:
Fixed and added corresponding unit test